cmds/core/shasum/shasum.go

```diff
@@ -20,18 +21,21 @@ var algorithm int
 
 var usage = "Usage:\nshasum -a <algorithm> <File Name>"
 
-// shaPrinter prints sha1/sha256 of given data. The
+// shaPrinter prints sha1/sha256/sha512 of given data. The
 // value of algorithm is expected to be 1 for SHA1
-// and 256 for SHA256
+// 256 for SHA256
+// and 512 for SHA512
 func shaGenerator(w io.Writer, r io.Reader, algo int) ([]byte, error) {
 	var h hash.Hash
 	switch algo {
 	case 1:
 		h = sha1.New()
 	case 256:
 		h = sha256.New()
+	case 512:
+		h = sha512.New()
 	default:
-		return nil, fmt.Errorf("invalid algorithm, only 1 or 256 are valid")
+		return nil, fmt.Errorf("invalid algorithm, only 1, 256 or 512 are valid")
```

Comment:
This patch could improve the basic code if you
fmt.Errorf("invalid algorithm, only 1, 256 or 512 are valid:%w", os.ErrInvalid)

Then you could change the test as below